src/orion/core/utils/format_trials.py

```diff
@@ -34,11 +34,17 @@ def tuple_to_trial(data, space):
     :type space: `orion.algo.space.Space`
     """
     assert len(data) == len(space)
-    params = [dict(
-        name=space[order].name,
-        type=space[order].type,
-        value=data[order]
-        ) for order in range(len(space))]
+    params = []
+    for i, dim in enumerate(space.values()):
+        try:
+            datum = data[i].tolist()  # if it is numpy.ndarray
+        except AttributeError:
+            datum = data[i]
+        params.append(dict(
+            name=dim.name,
+            type=dim.type,
+            value=datum
```

Comment:
But then trial_to_tuple will not have numpy arrays but lists. That could be problematic for algorithms. I believe there is a way to save numpy arrays in MongoDB. My laptop battery is about to die now. I'll look for this when I get home.

Reply:
I suppose that algorithms would like to have a way to **unpack** a `point in space` into a numpy.ndarray and **repack** a numpy.ndarray into a `point in space`. Points as they are shown in our test cases, are not favouring easy usage with numpy itself, but easy formatting them into the user script's input template. This is because a single point in a space with multiple Dimensions looks like: `(sp0, sp1, ..., spN)` where sp0 can be just a floating point, sp1 a flat list, and spN a numpy.array. Once you have the time, check out the two helper functions I introduced in bouthilx/orion.algo.skopt#3, they could be part of `orion.algo.base` I believe and their responsibility is to transform stuff into a single flat numpy array and back.

Reply:
ok so this problem occurs when we have a dimension with shape greater than 1. I would agree to migrate `pack_point` and `unpack_point` to `orion.algo.base`.

Still, I don't like the fact that if some_trial_tuple contains some numpy array, then the conversions diverges:
```python
some_trial_tuple != trial_to_tuple(tuple_to_trial(some_trial_tuple))
```
(abuse of notation with !=)

Reply:
Oh that's what you mean. Yeah I dislike this too.